src/Http/Controllers/ProvideToken.php

```diff
@@ -19,6 +19,8 @@ class ProvideToken
      */
     public function create(Request $request)
     {
+        abort_if(is_null($request->user()->otp_secret), 403, 'A secret is not set yet.');
```

Comment:
What purpose does this abort have? Maybe a mention in the view or a redirect to the setup if enabled would be a better choice?